core/src/test/java/hudson/util/io/ZipArchiverTest.java

```diff
@@ -99,7 +99,7 @@ public void huge64bitFile()  {
         File hugeFile = new File(tmpDir, "huge64bitFileTest.txt");
         try {
             RandomAccessFile largeFile = new RandomAccessFile(hugeFile, "rw");
-            largeFile.setLength(4 * 1024 * 1024 * 1024 + 2);
+            largeFile.setLength(4L * 1024 * 1024 * 1024 + 2);
```

Comment:
That's right, `System.out.println(4 * 1024 * 1024 * 1024 + 2);` prints 2
The file used is indeed 2 bytes lenght.

@oleg-nenashev maybe this is affecting here:
https://commons.apache.org/proper/commons-compress/javadocs/api-1.18/org/apache/commons/compress/archivers/zip/ZipArchiveOutputStream.html#setUseZip64-org.apache.commons.compress.archivers.zip.Zip64Mode-

> When setting the mode to AsNeeded, Zip64 extensions will transparently be used for those entries that require them. This mode can only be used if the uncompressed size of the ZipArchiveEntry is known when calling putArchiveEntry(org.apache.commons.compress.archivers.ArchiveEntry) or the archive is written to a seekable output (i.e. you have used the File-arg constructor) - this mode is not valid when the output stream is not seekable and the uncompressed size is unknown when putArchiveEntry(org.apache.commons.compress.archivers.ArchiveEntry) is called.